app/services/base_bookmarkable.rb

```diff
@@ -98,6 +98,33 @@ def self.reminder_handler(bookmark)
     raise NotImplementedError
   end
 
+  ##
+  # Can be used by the inheriting class via reminder_handler, most of the
+  # time we just want to make a Notification for a bookmark reminder, this
+  # gives consumers a way to do it without having provide all of the required
+  # data themselves.
+  #
+  # @param [Bookmark] bookmark          The bookmark that we are sending the reminder notification for.
+  # @param [Hash]     notification_data Any data, either top-level (e.g. topic_id, post_number) or inside
+  #                                     the data sub-key, which should be stored when the notification is
+  #                                     created.
+  # @return [void]
+  def self.send_reminder_notification(bookmark, notification_data)
+    if notification_data[:data].blank? ||
+        notification_data[:data][:bookmarkable_url].blank? ||
+        notification_data[:data][:title].blank?
+      raise Discourse::InvalidParameters.new("A `data` key must be present with at least `bookmarkable_url` and `title` entries.")
+    end
+
+    notification_data[:data] = (notification_data[:data] || {}).merge(
```

Comment:
Well technically speaking we don't need the `|| {}`

Reply:
Oh yes you are right, added that before the error check, will remove